content/4_Gold/Modular.mdx

```diff
@@ -479,6 +479,147 @@ int main() {
 
 </CPPOnly>
 
+# Chinese Remainder Theorem
+
+## For Two Congruences
+
+<Resources>
+	<Resource
+		  source="Stanford"
+		  title="The Chinese Remainder Theorem"
+		  url="https://crypto.stanford.edu/pbc/notes/numbertheory/crt.html"
+		  starred
+	>
+		In-depth explanation
+	</Resource>
+	<Resource
+		source="cp-algo"
+		title="Chinese Remainder Theorem"
+		url="https://cp-algorithms.com/algebra/chinese-remainder-theorem.html"
+	>
+	</Resource>
+	<Resource
+		source="CMU"
+		title="Chinese Remainder Theorem"
+		url="https://www.math.cmu.edu/~mradclif/teaching/127S19/Notes/ChineseRemainderTheorem.pdf"
+	>
+	</Resource>
+</Resources>
+
+<FocusProblem problem="CRT" />
+
+The [Chinese Remainder Theorem](https://en.wikipedia.org/wiki/Chinese_remainder_theorem) - also referred to as CRT - yields a unique
+solution to a system of simultaneous modular congruences with pairwise coprime moduli. More specifically, CRT determines a number $x$ that
+when divided by the given divisors leaves the given remainders. Mathematically, it solves the following system of modular congruences:
+
+$$
+x \equiv a_1 \pmod{m_1} \\
+x \equiv a_2 \pmod{m_2} \\
+x \equiv a_3 \pmod{m_3} \\
+\vdots\\
+x \equiv a_k \pmod{m_k} \\
+\tag{1}
+$$
+<center>$M=m_1\cdot m_2 \cdot m_3 \cdots m_k$</center>
+
+The above system has exactly one solution modulo $M$.
+
+We'll focus on the particular case for $k=2$; for two coprime moduli:
+
+$$
+x \equiv a_1 \pmod{m_1} \\
+x \equiv a_2 \pmod{m_2} \\
+\tag{2}
+$$
+<center>$M=m_1 \cdot m_2$</center>
+Let $n_1$ and $n_2 $be the modular inverse of $m_1$ modulo $m_2$, respecively, the modular inverse of $m_2$
+modulo $m_1$, thus the properties hold: $n_1 \cdot m_1 \equiv 1 \pmod{m_2}$ and $n_2 \cdot m_2 \equiv 1 \pmod{m_1}$.
+It's worth mentioning that $n_1$ and $n_2$ always exists because $\gcd(m_1,m_2)=1$. We can define a solution $x$ as:
+
+$$
+x = a_1 \cdot m_2 \cdot n_2 + a_2 \cdot m_1 \cdot n_1 \pmod{m_1\cdot m_2}
+$$
+
+It satisfies both equations:
+- Modulo $m_1$ we have $x \equiv a_1 \cdot m_2 \cdot n_2 \pmod{m_1} \iff x \equiv a_1 \pmod{m_1}$ since $m_2 \cdot n_2 \equiv 1 \pmod{m_1}$
+- Modulo $m_2$ we have $x \equiv a_2 \cdot m_1 \cdot n_1 \pmod{m_2} \iff x \equiv a_2 \pmod{m_2}$ since $m_1 \cdot n_1 \equiv 1 \pmod{m_2}$
+
+Now that we have a solution $x \pmod{M}$ it only remains to show that it is indeed the only valid
+solution modulo $m_1 \cdot m_2$. Assuming that there is a different valid solution $y$ we have: $m_1 | y-x$, $m_2 | y-x$,
+$m_3 | y-x$, $\ldots$, $m_k | y-x$, then it follows that $m_1 \cdot m_2 \cdots m_k$ divides $y-x$, since $m_1, m_2, \ldots, m_k$ are relatively
+prime. This means that: $y \equiv x \pmod{m_1 \cdot m_2 \cdots m_k}$.
```

Comment:
honestly i would just link to an external proof, it's been done so many times

Reply:
the resources have proof